vortex-serde/src/layouts/write/writer.rs

```diff
@@ -182,18 +183,102 @@ impl<W: VortexWrite> LayoutWriter<W> {
 }
 
 #[derive(Clone, Debug)]
-pub struct BatchOffsets {
+pub struct ColumnChunkAccumulator {
```

Comment:
I would actually split this up to avoid `Option<Scalar>` and instead store a field for each stat and take a generic type here for the value

Reply:
you just need `HashMap<Stat, Vec<Option<T>>` and `HashMap<Stat, Vec<Option<u64>>`